### PR TITLE
Add TypeScript declarations to paper-input.

### PR DESCRIPTION
**Pull request by aomarks**
This PR adds TypeScript declarations generated by https://github.com/Polymer/gen-typescript-declarations/

These declarations can be re-generated by running `npm run update-types`.

Tracker: https://github.com/Polymer/gen-typescript-declarations/issues/79

### CODE REVIEW

File: paper-input-addon-behavior.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    attached(): void;
+
+    /**
+     * The function called by `<paper-input-container>` when the input value or validity changes.
+     *
+     * @param state     inputElement: The input element.
+     *     value: The input value.
+     *     invalid: True if the input value is invalid.
+     */
+    update(state: {inputElement?: Element|null, value?: string, invalid: boolean}): void;
```
Comment (TimvdLippe) [inline paper-input-addon-behavior.d.ts]: Nit: put invalid first as that one is required.

File: paper-input-behavior.d.ts [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    /**
+     * If you're using PaperInputBehavior to implement your own paper-input-like
+     * element, bind this to the`<input is="iron-input">`'s `multiple` property,
+     * used with type=file.
+     */
+    multiple: boolean|null|undefined;
+    _ariaDescribedBy: string|null|undefined;
+    _ariaLabelledBy: string|null|undefined;
+    _inputId: string|null|undefined;
+    hostAttributes: object|null;
```
Comment (TimvdLippe) [inline paper-input-behavior.d.ts]: All these should be private

File: paper-input-behavior.d.ts [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+     * Returns a reference to the input element.
+     *      
+     */
+    readonly inputElement: any;
+
+    /**
+     * Returns a reference to the focusable element.
+     *      
+     */
+    readonly _focusableElement: any;
```
Comment (TimvdLippe) [inline paper-input-behavior.d.ts]: Should both be HTMLElement

File: paper-input-container.d.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  attached(): void;
+  _onAddonAttached(event: any): void;
+  _onFocus(): void;
+  _onBlur(): void;
+  _onInput(event: any): void;
+  _onValueChanged(event: any): void;
+  _handleValue(inputElement: any): void;
+  _handleValueAndAutoValidate(inputElement: any): void;
+  _onIronInputValidate(event: any): void;
+  _invalidChanged(): void;
```
Comment (TimvdLippe) [inline paper-input-container.d.ts]: Lots of methods which should be private I suppose? We can remove these later.

File: paper-input.d.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ * `--paper-input-container-ms-clear` | Mixin applied to the Internet Explorer reveal button (the eyeball) | {}
+ */
+interface PaperInputElement extends Polymer.Element, Polymer.PaperInputBehavior, Polymer.IronFormElementBehavior {
+
+  /**
+   * Returns a reference to the focusable element. Overridden from PaperInputBehavior
+   * to correctly focus the native input.
+   *      
+   */
+  readonly _focusableElement: any;
```
Comment (TimvdLippe) [inline paper-input.d.ts]: Can also be HTMLElement

File: paper-textarea.d.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+   */
+  rows: number|null|undefined;
+
+  /**
+   * The maximum number of rows this element can grow to until it
+   * scrolls. 0 means no maximum.
+   */
+  maxRows: number|null|undefined;
+  selectionStart: any;
+  selectionEnd: any;
```
Comment (TimvdLippe) [inline paper-textarea.d.ts]: Could be number.

Comment (aomarks) [reply paper-textarea.d.ts]: Done.

File: paper-input.d.ts (hunk 5 above)
Comment (aomarks) [reply paper-input.d.ts]: Done.

File: paper-input-container.d.ts (hunk 4 above)
Comment (aomarks) [reply paper-input-container.d.ts]: Done (some of them anyway).

File: paper-input-behavior.d.ts (hunk 2 above)
Comment (aomarks) [reply paper-input-behavior.d.ts]: Done.

File: paper-input-addon-behavior.d.ts (hunk 1 above)
Comment (aomarks) [reply paper-input-addon-behavior.d.ts]: Done.

**Pull request merged**